### PR TITLE
Update the ubuntu-cluster docs and scripts to 0.12.0

### PR DESCRIPTION
**Pull request by WIZARD-CXY**
Changelog
1 Users can configure the K8s version by changing some ENV in build.sh as the doc described.
2 Make the build.sh more robust to do some cleanup work after error
3 fix typos in docs and scripts

We will merge the ubuntu-cluster and ubuntu single node doc and script in the next PR.

### CODE REVIEW

**Comment by erictune** [comment]
Did you run the e2e test against your cluster?

**Comment by WIZARD-CXY** [comment]
not yet.I will test it today

**Comment by WIZARD-CXY** [comment]
May I ask how to run the e2e test and the expecting result? 
Many thanks!@erictune

**Comment by erictune** [comment]
You need a rebase because the README.md changed.

File: docs/getting-started-guides/ubuntu_multinodes_cluster.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ This document describes how to deploy kubernetes on multiple ubuntu nodes, inclu
 
 *3 These guide is tested OK on Ubuntu 14.04 LTS 64bit server, but it should also work on most Ubuntu versions*
 
-*4 Dependences of this guide: etcd-2.0.0, flannel-0.2.0, k8s-0.10.1, but it should also work on higher versions*
+*4 Dependences of this guide: etcd-2.0.0, flannel-0.2.0, K8s-0.12.0, but it should also work on higher versions*
```
Comment (erictune) [inline docs/getting-started-guides/ubuntu_multinodes_cluster.md]: may work with higher versions.

**Comment by WIZARD-CXY** [comment]
The ubuntu-cluster branch is some kind of broken in our side,I will send a new pull request using a new branch later

**Pull request closed without merging**